eng/scripts/live-test-resource-cleanup.ps1

```diff
@@ -3,30 +3,40 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
+# This script implements the resource management guidelines documented at https://github.com/Azure/azure-sdk-tools/blob/main/doc/engsys_resource_management.md
+
 #Requires -Version 6.0
 #Requires -PSEdition Core
+#Requires -Modules @{ModuleName='Az.Accounts'; ModuleVersion='1.6.4'}
+#Requires -Modules @{ModuleName='Az.Resources'; ModuleVersion='1.8.0'}
 
 [CmdletBinding(DefaultParameterSetName = 'Default', SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
 param (
-    [Parameter(Mandatory = $true)]
+    [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
     [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
     [string] $ProvisionerApplicationId,
 
-    [Parameter(Mandatory = $true)]
+    [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
     [string] $ProvisionerApplicationSecret,
 
-    [Parameter(Mandatory = $true)]
+    [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
     [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
-    [ValidateNotNullOrEmpty()]
     [string] $TenantId,
 
-    [Parameter(Mandatory = $true)]
+    [Parameter(ParameterSetName = 'Provisioner', Mandatory = $false)]
+    [ValidateNotNullOrEmpty()]
+    [string] $Environment = "AzureCloud",
+
+    [Parameter(Mandatory = $false)]
     [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
     [string] $SubscriptionId,
 
     [Parameter()]
-    [ValidateNotNullOrEmpty()]
-    $Environment = "AzureCloud",
+    [int] $DeleteAfterHours = 24,
+
+    [Parameter()]
```

Comment:
Should this be added to the "Default" parameter set? That's not actually a "default" but just a placeholder. Technically, something like "__AllParameterSets" - a reserved name - is. It's just a convention I use for "not Provisioner" (or whatever). But it seems that specifying a provisioner or login are mutually exclusive, yes? You could also make up a new parameter set name and make that the default in the `[CmdletBinding()]` attribute.